BJS2-52594 : Написание рекомендации

MaksimMas opened the pull request.

MaksimMas committed:
Проверка тестов

MaksimMas committed:
Проверка тестов

MaksimMas committed:
Проверка тестов

Timka-cloud1 commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,98 @@
+    }
+
+    void validateOfSkills(List<SkillOfferDto> skills) {
+        List<Long> skillIds = skills.stream()
```
было бы круто сделать такие методы приватными, так как это внутренняя кухня самого сервиса и извне не давать их вызывать

AlexeyAntipin commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,98 @@
+                .toList();
+    }
+
+    RecommendationDto saveSkillOffers(RecommendationDto recommendation) {
```
Не стоит оставлять методы в бизнес логике без модификатора доступа

AlexeyAntipin commented on src/test/java/school/faang/user_service/service/RecommendationControllerTest.java:
```diff
@@ -0,0 +1,55 @@
+    @Test
+    public void testEmptyContentIsInvalid() {
+        assertThrows(IllegalArgumentException.class,
+                () -> recommendationController.giveRecommendation(recommendationDtoTestEmpty));
```
Все тесты написаны только на 1 метод, остальные 3 метода в контроллере не протестированы

AlexeyAntipin commented:
И можешь ещё, пожалуйста, назвать PR следующим образом: <Номер задачи> | <Название задачи>. ЧТобы это потом было понятно в истории коммитов

MaksimMas committed:
1. Доделал тесты
2. Убрал проверку данных из Сервиса в другой пакет для упрощения написания тестов
3. Создал свой Exception
